src/osd/OSD.cc

```diff
@@ -6266,6 +6266,31 @@ void OSD::tick()
     }
   }
 
+  int map_cache_max_size = service.map_cache.get_max_size();
+  if (map_cache_max_size > cct->_conf->osd_map_cache_size) {
+    epoch_t osd_min = 0;
+    for (auto shard : shards) {
+      epoch_t min = shard->get_min_pg_epoch();
+      if (osd_min == 0 || min < osd_min) {
+        osd_min = min;
+      }
+    }
+    if (osd_min == superblock.get_newest_map()) {
+      map_cache_max_size -= cct->_conf->osd_map_cache_reduce_step;
+      if (map_cache_max_size < cct->_conf->osd_map_cache_size) {
+        map_cache_max_size = cct->_conf->osd_map_cache_size;
+      }
+      service.map_cache.set_size(map_cache_max_size);
```

Comment:
I'm not sure that this commit worth the added complexity when there is an already existing configurable.
A user can simply set `osd_map_cache_size` to 500 or to 1024 if the intent is to increase performance over memory usage.